aws-toolkit-eclipse-master/bundles/com.amazonaws.eclipse.sdk.ui/samples/AmazonKinesis/AmazonKinesisRecordProducerSample.java

```diff
@@ -128,18 +128,19 @@ public static void main(String[] args) throws Exception {
         System.out.printf("Putting records in stream : %s until this application is stopped...\n", myStreamName);
         System.out.println("Press CTRL-C to stop.");
         // Write records to the stream until this program is aborted.
+        PutRecordsRequest putRecordsRequest  = new PutRecordsRequest();
+        putRecordsRequest.setStreamName(myStreamName);
+        List <PutRecordsRequestEntry> putRecordsRequestEntryList  = new ArrayList<>();        
         while (true) {
             long createTime = System.currentTimeMillis();
             PutRecordRequest putRecordRequest = new PutRecordRequest();
-            putRecordRequest.setStreamName(myStreamName);
             putRecordRequest.setData(ByteBuffer.wrap(String.format("testData-%d", createTime).getBytes()));
-            putRecordRequest.setPartitionKey(String.format("partitionKey-%d", createTime));
-            PutRecordResult putRecordResult = kinesis.putRecord(putRecordRequest);
-            System.out.printf("Successfully put record, partition key : %s, ShardID : %s, SequenceNumber : %s.\n",
-                    putRecordRequest.getPartitionKey(),
-                    putRecordResult.getShardId(),
-                    putRecordResult.getSequenceNumber());
+            putRecordRequest.setPartitionKey(String.format("partitionKey-%d", createTime));            
+            putRecordsRequestEntryList.add(putRecordRequest); 
         }
+        putRecordsRequest.setRecords(putRecordsRequestEntryList);
+        PutRecordResult putRecordResult = kinesis.putRecords(putRecordsRequest);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: There may be failed records in the output of the [PutRecords ](https://docs.aws.amazon.com/kinesis/latest/APIReference/API_PutRecords.html)operation. This may lead to loss of records.

**Fix**: Check the FailedRecordCount parameter in the putRecordsResult to confirm if there are failed records in the request. If there are failed records, each putRecordsEntry that has a non-null ErrorCode should be added to a subsequent request.

[Learn more](https://docs.aws.amazon.com/streams/latest/dev/developing-producers-with-sdk.html#kinesis-using-sdk-java-putrecords)